litestar/plugins/problem_details.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
```

Comment:
Is it a problem that the current docs use application/problem+json as an example for json encoding when there is a plugin for it? https://docs.litestar.dev/2/usage/responses.html#json-responses

It might be fine? Technically there is nothing incorrect, it just might give someone the impression that that is the way to handle problem details, so they won't find the plugin.

Reply:
Yupp, we'll link to the plugin from there instead.

Reply:
Does that make sense? The example isn't about how to use problem details, but how to use "+json" suffixed media types. Would it make more sense to change the example to use some arbitrary "+json" suffixed media type?

Reply:
Good point. Is there another common usecase of the suffix? If not, I was thinking we could keep that, but then also note that there is a builtin plugin for this specific case.

Reply:
Okay, I just used a generic example for this instead of problem details.